src/clients/bridge.ts

```diff
@@ -0,0 +1,71 @@
+import { TextChannel } from 'discord.js';
+import { linkedListNode, nodeInterface } from '../linkedList';
+import { twitchMsg } from '../messageObjects';
+import tmijs from 'tmi.js';
+import configFile from '../../config.json';
+
+/**
+ * If an error happens either on twitch or discord, print the thing
+ * @param {*} error 
+ */
+export const genericPromiseError = (error: any) => console.error('Snap, I hit a snag... >.<', error);
+
+interface IBridge
+{
+    MAX_MSG_CACHE: number;
+    currMsgCount: number;
+    targetDiscordChannel: TextChannel | undefined,
+    twitchClient: tmijs.Client | undefined,
+    discordTwitchCacheMap: Map<any, any>
+    twitchMessageSearchCache: { [key: string]: linkedListNode; };
+    messageLinkdListInterface: nodeInterface;
+    lastUserStateMsg: any
+}
+
+const Bridge: IBridge = {
```

Comment:
Ok, question from a typescript newbie - is it necessary to have the interface for the bridge? Seems to take up a lot of space, and I wonder if you could just specify the types themselves within the object definition (my understanding is that it's possible). I guess another question is would that even look elegent :/

Reply:
It does look ugly, but it works! Although we could do something like this (see block) if you don't like having an interface for it; although I made the interface cause it's nice to have cause funny types. 
```ts
const obj = {
     thing: 'text' as String,
     thing2: 2 as number
}```

Reply:
ooh, looks aesthetically pleasing! I might have weird taste though

On my side that's preferable, but I trust your judgement if the interface helps with something like intellisense

Reply:
Either way is fine, same with not having any; the properties on the object will automatically be inferred although I like using an interface because if you remove or add a property that's not on the interface it'll throw an error; the way above is just implicitly setting the type. Any way is fine, up to you Mr. project man!

Reply:
gotcha, makes sense 👍 

Generally I'm kindof a minimalist when it comes to writing code; if intellisense can help with working with this object, I say it probably makes sense to axe the interface. Further we are only instantiating the object once, so we wouldn't need to ensure the contents all the time if we can reference that from this file

Reply:
Removed interface, now using casting; Resolving this cause it's done.